xwiki-platform-core/xwiki-platform-observation/xwiki-platform-observation-remote/xwiki-platform-observation-remote-api/src/main/java/org/xwiki/observation/remote/converter/DocumentRemoteEventConverter.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.observation.remote.converter;
+
+import java.io.Serializable;
+import java.util.Hashtable;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.bridge.DocumentModelBridge;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Component helper for serializing standard XWiki elements such as {@code XWikiDocument} or {@code XWikiContext}.
+ *
+ * @version $Id$
+ * @since 15.2RC1
+ */
+@Unstable
+@Role
+public interface DocumentRemoteEventConverter
+{
+    /**
+     * Allow to serialize an {@code XWikiContext}. The argument should be an actual instance of
+     * {@code com.xpn.xwiki.XWikiContext}.
+     *
+     * @param context the context to be serialized
+     * @return a {@link Serializable} object to be used in a remote event
+     */
+    Serializable serializeXWikiContext(Hashtable<Object, Object> context);
```

Comment:
Hm so you suggest to remove this API from there, and to document to use `ContextStoreManager#save` which would return a serializable map instead? Do you also suggest to change the current implem of `AbstractXWikiEventConverter` for the serialization of contexts?

Reply:
> Hm so you suggest to remove this API from there, and to document to use ContextStoreManager#save which would return a serializable map instead?

Yes, each converter would then choose which types of context info it would like to keep.

> Do you also suggest to change the current implem of AbstractXWikiEventConverter for the serialization of contexts?

That would probably be better, yes. More future-proof than playing with XWikiContext directly.

Reply:
ok I'll check to perform those changes then